### PR TITLE
make compaction not duplicate data

### PR DESCRIPTION
**Pull request by LtChae**
smartcitiesdata/smartcitiesdata#507

co-authored-by: Benjamin Brewer <benjamin.brewer@accenture.com>

### CODE REVIEW

File: apps/forklift/lib/forklift/message_handler.ex [hunk 1]
```diff
@@ -21,11 +21,14 @@ defmodule Forklift.MessageHandler do
...
-    |> Enum.reject(&error_tuple?/1)
-    |> Forklift.DataWriter.write(dataset: dataset)
+    timed_messages =
+      messages
+      |> Enum.map(&parse/1)
+      |> Enum.map(&yeet_error/1)
+      |> Enum.reject(&error_tuple?/1)
+      |> Forklift.DataWriter.write(dataset: dataset)
+
+    Task.start(fn -> Forklift.DataWriter.write_to_topic(timed_messages) end)
```
Comment (mcsearchin) [inline apps/forklift/lib/forklift/message_handler.ex]: Is there anything we should be doing once this task is complete? Even just debug logging the end result?

**Comment by mcsearchin** [comment]
It's not clear from looking at the diff... are there any tests here that verify that we don't duplicate data during compaction?  I'm guessing this is the kind of test that wouldn't have a strong JSR - just curious.

**Comment by bbrewer-papa-x** [comment]
> It's not clear from looking at the diff... are there any tests here that verify that we don't duplicate data during compaction? I'm guessing this is the kind of test that wouldn't have a strong JSR - just curious.

We considered putting a count check into the geojson part of the e2e test. I can give that a shot and see if it's stable. That test has always been a bit flaky though

File: apps/forklift/lib/forklift/message_handler.ex (hunk 1 above)
Comment (bbrewer-papa-x) [reply apps/forklift/lib/forklift/message_handler.ex]: We're putting a debugge message in there

**Pull request merged**